tests/drivers/gnss/gnss_api/src/test_enabled_systems.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gnss.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/ztest.h>
+
+#define TEST_SEARCH_PERIOD K_SECONDS(CONFIG_TEST_SEARCH_PERIOD)
+
+static const struct device *dev = DEVICE_DT_GET(DT_ALIAS(gnss));
+static const gnss_systems_t enabled_systems_array[] = {
+	CONFIG_TEST_ENABLED_SYSTEMS_0,
+	CONFIG_TEST_ENABLED_SYSTEMS_1,
+	CONFIG_TEST_ENABLED_SYSTEMS_2,
+	CONFIG_TEST_ENABLED_SYSTEMS_3,
+};
+
+static bool test_reported_are_expected(gnss_systems_t reported, gnss_systems_t expected)
+{
+	return ((~expected) & reported) == 0;
+}
+
+static void test_set_enabled_systems(gnss_systems_t enabled_systems)
+{
+	int ret;
+
+	ret = gnss_set_enabled_systems(dev, enabled_systems);
+	zassert_ok(ret, "failed to set enabled systems (%i)", ret);
+}
+
+static void test_get_system_enabled(gnss_systems_t expected_systems)
+{
+	int ret;
+	gnss_systems_t enabled_systems;
+
+	ret = gnss_get_enabled_systems(dev, &enabled_systems);
+	if (ret == -ENOSYS) {
+		return;
+	}
+	zassert_ok(ret, "failed to get enabled systems (%i)", ret);
+
+	zassert_equal(enabled_systems, expected_systems,
+		      "invalid enabled systems (%u != %u)",
+		      enabled_systems, expected_systems);
+}
+
+#ifdef CONFIG_GNSS_SATELLITES
+static atomic_t reported_systems_atom = ATOMIC_INIT(0);
+
+static void gnss_satellites_cb(const struct device *dev,
+			       const struct gnss_satellite *satellites,
+			       uint16_t size)
+{
+	for (uint16_t i = 0; i < size; i++) {
+		atomic_or(&reported_systems_atom, satellites[i].system);
+	}
+}
+
+GNSS_SATELLITES_CALLBACK_DEFINE(DEVICE_DT_GET(DT_ALIAS(gnss)), gnss_satellites_cb);
+
+static void test_validate_satellites(gnss_systems_t expected_systems)
+{
+	gnss_systems_t reported_systems;
+	bool expected;
+
+	atomic_set(&reported_systems_atom, 0);
+	PRINT("searching with enabled system %u\n", expected_systems);
+	k_sleep(TEST_SEARCH_PERIOD);
```

Comment:
Is it no problem to sleep 60 seconds in tests (moreover multiple times)?

Reply:
Yes, I ran the tests locally where its fine, but twister/CI will time out after 60 seconds by default. Upped timeout to 300s (systems enabled test should take 240s) in testcase.yaml